src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+class FigureSupplier {
+    private ColorSupplier colorSupplier;
+
+    public FigureSupplier(ColorSupplier colorSupplier) {
+        this.colorSupplier = colorSupplier;
+    }
+
+    public Figure getRandomFigure() {
+        Random random = new Random();
+        int figureType = random.nextInt(5);
+
+        double side = random.nextDouble() * 20 + 1;
+        double length = random.nextDouble() * 20 + 1;
+        double width = random.nextDouble() * 20 + 1;
+        double firstLeg = random.nextDouble() * 20 + 1;
+        double secondLeg = random.nextDouble() * 20 + 1;
+        double radius = random.nextDouble() * 20 + 1;
+        double topBase = random.nextDouble() * 20 + 1;
+        double bottomBase = random.nextDouble() * 20 + 1;
+        double legs = random.nextDouble() * 20 + 1;
+
+        String color = colorSupplier.getRandomColor();
+
+        switch (figureType) {
+            case 0:
+                return new Square(side, color);
```

Comment:
If you create method you will be able to simplify that with:
```
return new Square(getRandomValue(), colorSupplier.getRandomColor());
```
And remove temporary variables.